app/controllers/test_reports_controller.rb

```diff
@@ -2,29 +2,27 @@
 
 class TestReportsController < ApplicationController
   before_action :check_round, only: %i[show update]
-
   def index
     per_page = 30
-    @jobs = Job.all
-               .page(params[:page])
-               .per(per_page)
-               .order('jobs.id DESC')
-    job_ids = Job.select('id').page(params[:page]).per(per_page).order('jobs.id DESC').to_a
-    @joined_jobs = Job.join_with_suites
-                      .select_group_concat_suites
-                      .group('jobs.id')
-                      .where(id: job_ids)
-                      .order('jobs.id DESC')
+    @page = Kaminari.paginate_array(Job.all_root_jobs.to_a, total_count: Job.all_root_jobs.length).page(params[:page]).per(per_page)
+    start_num = params[:page].nil? || params[:page] == 1 ? 0 : per_page * (params[:page].to_i - 1)
+    root_jobs = Job.root_jobs(start_num, per_page)
+    # I guessed the number of chiled_jobs per page is obtained by per_page*4
```

Comment:
It might be a good idea to delete subject.
```suggestion
    # Guessed the number of chiled_jobs per page is obtained by per_page*4
```